curated_transformers/tokenization/_fairseq.py

```diff
@@ -0,0 +1,91 @@
+from typing import Callable, Iterable, List
+
+from .tokenizer import PiecesWithIds, PostEncoder, PreDecoder
+from .util import remove_pieces_from_sequence, add_bos_eos_to_encoding
+
+
+class FAIRSEQ_PIECE_IDS:
+    FAIRSEQ_BOS = 0
+    FAIRSEQ_EOS = 2
+    FAIRSEQ_UNK = 3
+
+    SPP_BOS = 1
+    SPP_EOS = 2
+    SPP_UNK = 0
+
+
+class FairSeqPostEncoder(PostEncoder):
+    """Performs fixups of SentencePiece piece identifiers for models that use
+    the fairseq vocabulary."""
+
+    def __init__(
+        self,
+        *,
+        bos_piece: str,
+        eos_piece: str,
+        bos_id: int,
+        eos_id: int,
+        piece_updater: Callable[[int], int],
+    ):
+        """Construct a fairseq post-encoder.
+
+        bos_piece (str): The piece used to mark the beginning of a sequence.
```

Comment:
When adding new public classes, we should probably immediately do the documentation Sphinx-style and wire the classes up in `docs/`.